cmd/kpod/run.go

```diff
@@ -85,20 +86,37 @@ func runCmd(c *cli.Context) error {
 		libpod.WriteFile(ctr.ID(), c.String("cidfile"))
 		return nil
 	}
+
+	// Create a bool channel to track that the console socket attach
+	// is sucessful.
+	attached := make(chan bool)
+	// Create a waitgroup so we can sync and wait for all goroutines
+	// to finish before exiting main
+	var wg sync.WaitGroup
+
+	if !createConfig.detach {
+		// We increment the wg counter because we need to do the attach
+		wg.Add(1)
+		// Attach to the running container
+		go func() {
+			logrus.Debug("trying to attach to the container %s", ctr.ID())
+			defer wg.Done()
+			if err := ctr.Attach(false, c.String("detach-keys"), attached); err != nil {
+				logrus.Errorf("unable to attach to container %s: %q", ctr.ID(), err)
+			}
+		}()
+		if !<-attached {
+			return errors.Errorf("unable to attach to container %s", ctr.ID())
+		}
+	}
 	// Start the container
 	if err := ctr.Start(); err != nil {
 		return errors.Wrapf(err, "unable to start container %q", ctr.ID())
 	}
 	logrus.Debug("started container ", ctr.ID())
-	if createConfig.tty {
-		// Attach to the running container
-		logrus.Debug("trying to attach to the container %s", ctr.ID())
-		if err := ctr.Attach(false, c.String("detach-keys")); err != nil {
-			return errors.Wrapf(err, "unable to attach to container %s", ctr.ID())
-		}
-	} else {
-		fmt.Printf("%s\n", ctr.ID())
-	}
-
+	//else {
+	//	fmt.Printf("%s\n", ctr.ID())
```

Comment:
Shouldn't this happen if you specify -d?

Reply:
Yes, this should still happen

Reply:
fixed